### PR TITLE
AGENT-121: Change default docker mode to raw-logs

### PR DESCRIPTION
**Pull request by echee2**
The current docker agent defaults to the legacy `docker-api` mode whereby raw logs mounted on the agent container are not directly monitored, rather stdout/err logs are gotten via docker api, written to disk once more, and those files monitored.

We originally kept the legacy mode as default in order not to impact existing customers. However, that approach is not only suboptimal but the docker api is not 100% reliable. 

The better approach is to directly monitor mounted other-container logs (raw logs mode). 

This PR changes the default mode to that and also emits a warning whenever a mounted container log file is missing (e.g. in the case where customer fails to mount the docker storage directory per instructions).

# How tested?

Manually: 

1. ensured `Starting docker monitor (raw_logs=True)` is emitted in agent.log.
2. purposefully omit the mounting of /var/lib/docker/containers, then verified the warning is 
emitted.

### CODE REVIEW

File: docker/docker-json-config/agent.d/docker.json [hunk 1]
```diff
@@ -3,10 +3,9 @@
 // docker container log directory (which must be mounted into the
 // agent's container).
 {
- monitors: [
-   {
-    module: "scalyr_agent.builtin_monitors.docker_monitor",
-    docker_raw_logs: true
```
Comment (echee2) [inline docker/docker-json-config/agent.d/docker.json]: Please pay attention to the removal of this line -- specifically whether this will impact existing customers?

Comment (echee2) [inline docker/docker-json-config/agent.d/docker.json]: Safe to remove? Does it impact existing customers?

Comment (imron) [reply docker/docker-json-config/agent.d/docker.json]: It won't impact existing customers.  Only customers who use the new version will get this config file, and for them the default will be true.

File: scalyr_agent/builtin_monitors/docker_monitor.py [hunk 2]
```diff
@@ -508,6 +508,7 @@ def start( self ):
         #create and start the DockerLoggers
         self.__start_docker_logs( self.docker_logs )
         self._logger.log(scalyr_logging.DEBUG_LEVEL_1, "Initialization complete.  Starting docker monitor for Scalyr" )
```
Comment (imron) [inline scalyr_agent/builtin_monitors/docker_monitor.py]: I think you can remove this DEBUG_LEVEL_1 log now that the following line prints almost the same message.

File: scalyr_agent/builtin_monitors/docker_monitor.py [hunk 3]
```diff
@@ -101,7 +101,7 @@
                      'Optional (defaults to False). If True, the docker monitor will use the raw log files on disk to read logs.'
```
Comment (imron) [inline scalyr_agent/builtin_monitors/docker_monitor.py]: Need to update the comment that says it defaults to `False`.

File: scalyr_agent/builtin_monitors/docker_monitor.py [hunk 4]
```diff
@@ -707,6 +708,10 @@ def __start_docker_logs( self, docker_logs ):
                     global_log.info( "Error adding log '%s' to log watcher - %s" % (log['log_config']['path'], e) )
 
             if self._use_raw_logs:
+                log_path = log['log_config']['path']
+                if not os.path.exists(log_path):
+                    global_log.warn("Missing file detected for container log path '%s'. Please ensure that "
+                                    "/var/lib/docker/containers (or custom storage directory) is mounted." % log_path)
```
Comment (imron) [inline scalyr_agent/builtin_monitors/docker_monitor.py]: 'custom storage directory' is not so clear.  I would reword the second sentence to something like:

"Please ensure that the host's Docker container directory (by default `/var/lib/docker/containers`) has been mounted in the Scalyr Agent container."
